src/cli/lib/src/command_runner.dart

```diff
@@ -0,0 +1,149 @@
+import 'package:args/args.dart';
+import 'package:args/command_runner.dart';
+import 'package:cli_completion/cli_completion.dart';
+import 'package:flutter_application_generator/src/commands/create_command.dart';
+import 'package:flutter_application_generator/src/commands/update_command.dart';
+import 'package:flutter_application_generator/src/version.g.dart';
+import 'package:mason_logger/mason_logger.dart';
+import 'package:pub_updater/pub_updater.dart';
+
+const executableName = 'flutter_application_generator';
+const packageName = 'flutter_application_generator';
+const description = 'A Flutter application template with built-in recipes.';
+
+/// {@template flutter_application_generator_command_runner}
+/// A [CommandRunner] for the CLI.
+///
+/// ```
+/// $ flutter_application_generator --version
+/// ```
+/// {@endtemplate}
+final class FlutterApplicationGeneratorCommandRunner
+    extends CompletionCommandRunner<int> {
+  final Logger _logger;
+  final PubUpdater _pubUpdater;
+
+  /// {@macro flutter_application_generator_command_runner}
+  FlutterApplicationGeneratorCommandRunner({
+    Logger? logger,
+    PubUpdater? pubUpdater,
+  })  : _logger = logger ?? Logger(),
+        _pubUpdater = pubUpdater ?? PubUpdater(),
+        super(executableName, description) {
+    // Add root options and flags.
+    argParser
+      ..addFlag(
+        'version',
+        abbr: 'v',
+        negatable: false,
+        help: 'Print the current version.',
+      )
+      ..addFlag(
+        'verbose',
+        help: 'Noisy logging, including all shell commands executed.',
+      );
+
+    // Add sub commands.
+    addCommand(CreateCommand(logger: _logger));
+    addCommand(UpdateCommand(logger: _logger, pubUpdater: _pubUpdater));
+  }
+
+  @override
+  void printUsage() => _logger.info(usage);
+
+  @override
+  Future<int> run(Iterable<String> args) async {
+    try {
+      final topLevelResults = parse(args);
+      if (topLevelResults['verbose'] == true) {
+        _logger.level = Level.verbose;
+      }
+      return await runCommand(topLevelResults) ?? ExitCode.success.code;
+    } on FormatException catch (e, stackTrace) {
+      // On format errors, show the commands error message, root usage and
+      // exit with an error code.
+      _logger
+        ..err(e.message)
+        ..err('$stackTrace')
+        ..info('')
+        ..info(usage);
+      return ExitCode.usage.code;
+    } on UsageException catch (e) {
+      // On usage errors, show the commands usage message and
+      // exit with an error code.
+      _logger
+        ..err(e.message)
+        ..info('')
+        ..info(e.usage);
+      return ExitCode.usage.code;
+    }
+  }
+
+  @override
+  Future<int?> runCommand(ArgResults topLevelResults) async {
+    // Fast track completion command.
+    if (topLevelResults.command?.name == 'completion') {
```

Comment:
I'm not sure to understand when this could happen. Like are we referring to this CLI can be run more than once in parallel?

Reply:
Think they talk about it here.